encrypted/session.go

```diff
@@ -286,10 +288,17 @@ func (info *sessionInfo) _handleUpdate(init *sessionInit) {
 	info._resetTimer()
 }
 
+var bufPool = sync.Pool{
+	New: func() interface{} {
+		return make([]byte, 0, sessionTrafficOverhead+network.MaxPeerMessageSize)
```

Comment:
Instead of importing `network`, I would prefer to depend on just the interface type and call `MTU()` (which is how related things are handled in `signed` and in ygg), since that lets things layer on top of each other a little more cleanly. That would (I think) also involve moving the pool into the struct, instead of being a package global. Alternatively, we just don't try to guess the size and preallocate, just return a nil slice instead. Go will normally allocate once, when we go to use the slice, and since most packets tend to be MTU size anyway, I wouldn't expect it to hurt anything in practice.